src/python/pants/base/exceptions.py

```diff
@@ -62,3 +62,28 @@ class BackendConfigurationError(BuildConfigurationError):
 
 class IncompatiblePlatformsError(Exception):
   """Indicates that target platforms are incompatible with a target that contains native code."""
+
+
+class GracefulTerminationException(Exception):
```

Comment:
It's not clear why TaskError above with exit_code nonzero doesn't fit the bill. Granted there is the name and the fact it also carries failed_targets. Perhaps exceptions bound for the v2 rule world should live in a different file / package to help make it clear we're moving to new names? It really seems like this should live in pants.engine.rules.

Reply:
Moved into src/python/pants/rules/core/exceptions.py - happy to re-use `TaskError` if that feels better to folks, but I quite like single-purpose exceptions for this kind of context